docs/concepts/time-and-windows-in-ksql-queries.rst

```diff
@@ -0,0 +1,305 @@
+    time windows. For example, if you define a 24-hour tumbling time window, it will
+    be in the UTC timezone, which may not be appropriate if you want to have daily
+    windows in your timezone.
+
+Window Types
+============
+
+There are three ways to define time windows in KSQL.
+
+.. image:: ../img/ksql-window-aggregation.png
+   :alt: Diagram showing three types of time windows in KSQL streams: tumbling, hopping, and session
+
+Tumbling Window
+---------------
+
+All time windows are the same size and adjacent to each other, which
+means that whenever a window ends, the next window starts.
+
+For example, if you want to compute the the five highest-value orders
+per zip code per hour in an ``orders`` stream, you might run a query like this:
+
+.. code:: sql
+
+    SELECT orderzip_code, TOPK(order_total, 5) FROM orders
+      WINDOW TUMBLING (SIZE 1 HOUR) GROUP BY order_zipcode;
+
+Here's another example: to detect potential credit card fraud in an
+``authorization_attempts`` stream, you might run a query for the number of
+authorization attempts on a particular card that's greater than three, during
+a time interval of five seconds.
+
+.. code:: sql
+
+    SELECT card_number, count(*) FROM authorization_attempts
+      WINDOW TUMBLING (SIZE 5 SECONDS)
+      GROUP BY card_number HAVING count(*) > 3;
+
+The tumbling window's start time is inclusive, but the end time is exclusive.
+This is important for non-overlapping windows, in which each record must be
+contained in exactly one window.
+
+Hopping Window
+--------------
+
+All time windows are the same size, but they might overlap, depending on the
+length of time specified in the ADVANCE BY property.
+
+For example, if you want to count the pageviews for only ``Region_6`` by female
+users for a hopping window of 30 seconds that advances by 10 seconds, you might
+run a query like this:
+
+.. code:: sql
+
+    SELECT regionid, count(*) FROM pageviews 
+      WINDOW HOPPING (SIZE 30 SECONDS, ADVANCE BY 10 SECONDS)
+      WHERE UCASE(gender)='FEMALE' AND LCASE (regionid) LIKE '%_6'
+      GROUP BY regionid;
+
+The hopping window's start time is inclusive, but the end time is exclusive.
+This is important for non-overlapping windows, in which each record must be
+contained in exactly one window.
+
+Session Window
+--------------
+
+A new window starts if the last event that arrived is further back in time
+than a specified session timeout time.
+
+For example, to count the number of pageviews per region for session windows
+with a session inactivity gap of 60 seconds, you might run the following query,
+which "sessionizes" the input data and performs the counting/aggregation step
+per region:
+
+.. code:: sql
+
+    SELECT regionid, count(*) FROM pageviews
+      WINDOW SESSION (60 SECONDS)
+      GROUP BY regionid;
+
+The start and end times for a session window are both inclusive, in contrast to
+time windows. There is always a record in the session window with both the start

```

Comment:
Sounds confusing: "There is always a record in the session window with both the start and end timestamps, [...]"

What do you want to say here?

I think it's about:

- A session window contains at least one record.  It's not possible for a session window to have zero records.
- If a session window contains exactly 1 record, then that record's timestamp (via ROWTIME) is identical to the window's own start and end times (which the user can access via WINDOWSTART() and WINDOWEND(), respectively).
- If a session window contains 2 or more records, then the "earliest/oldest" record's timestamp (ROWTIME) is identical to the window's start time (WINDOWSTART()), and the "latest/newest" record's timestamp (ROWTIME) is identical to the window's end time (WINDOWEND()).

Is the above correct?